linode/helper/frameworkfilter/filter.go

```diff
@@ -45,6 +45,12 @@ type FilterAttribute struct {
 
 	// Converts the filter string to the correct type.
 	TypeFunc FilterTypeFunc
+
+	// (Optional) Allows this attribute during validation
+	// for the order and order_by fields.
+	// This is useful for edge cases where we want order
+	// through the API but filter on the client.
+	AllowOrder bool
```

Comment:
Can we always allow order so this flag won't be necessary? Any case ordering is not allowed or not appropriated?

Reply:
I added this flag so we could keep the client-side `order_by` field validation at plan-time to prevent any unexpected failures during apply. It might be a good idea to defer this logic to the API, though.

Reply:
So for the existing data source/resource without `AllowOrder`, we should add them to their filter framework config?

Reply:
I might need to adjust the name to make the meaning of this field a bit more clear, but right now it only affects non-filterable fields that can be ordered on. I don't think we'll need to make changes to other resources regardless though 👍